include/drivers/uart.h

```diff
@@ -467,343 +484,47 @@ __subsystem struct uart_driver_api {
 
 };
 
+/** @endcond */
 
 /**
- * @brief Set event handler function.
- *
- * Since it is mandatory to set callback to use other asynchronous functions,
- * it can be used to detect if the device supports asynchronous API. Remaining
- * API does not have that detection.
+ * @brief Check whether an error was detected.
  *
- * @param dev       UART device structure.
- * @param callback  Event handler.
- * @param user_data Data to pass to event handler function.
+ * @param dev UART device instance.
  *
- * @retval -ENOSYS  If not supported by the device.
- * @retval -ENOTSUP If API not enabled.
- * @retval 0	    If successful, negative errno code otherwise.
+ * @retval 0 If no error was detected.
+ * @retval err Error flags as defined in @ref uart_rx_stop_reason
```

Comment:
I think it should actually be:
```suggestion
 * @retval err Error flags as defined in #uart_rx_stop_reason
```

https://www.doxygen.nl/manual/autolink.html

Reply:
Mhh, `@ref` works fine, it renders the reference correctly. It is used all over the place in this file (53 times), so I'd prefer to keep it.

Reply:
it's ok (mentioned in case it didn't resolve as I've had to use `#` sometimes).